scripts/build/termux_create_debian_subpackages.sh

```diff
@@ -1,27 +1,21 @@
 termux_create_debian_subpackages() {
-	# Sub packages:
-	local _ADD_PREFIX=""
-	if [[ "$TERMUX_PACKAGE_LIBRARY" == 'glibc' ]]; then
-		_ADD_PREFIX="glibc/"
-	fi
-	if [[ "$TERMUX_PKG_NO_STATICSPLIT" == 'false' && -n "$(shopt -s globstar; shopt -s nullglob; echo ${_ADD_PREFIX}lib/**/*.a)" ]]; then
-		# Add virtual -static sub package if there are include files:
-		local _STATIC_SUBPACKAGE_FILE=$TERMUX_PKG_TMPDIR/${TERMUX_PKG_NAME}-static.subpackage.sh
-		echo TERMUX_SUBPKG_INCLUDE=\"$(find ${_ADD_PREFIX}lib -name '*.a' -o -name '*.la') $TERMUX_PKG_STATICSPLIT_EXTRA_PATTERNS\" > "$_STATIC_SUBPACKAGE_FILE"
-		echo "TERMUX_SUBPKG_DESCRIPTION=\"Static libraries for ${TERMUX_PKG_NAME}\"" >> "$_STATIC_SUBPACKAGE_FILE"
-	fi
+	local TERMUX_PKG_FILE TERMUX_PKG_ARCH TERMUX_PARENT_DEPEND_ON_SUBPKG
 
 	# Now build all sub packages
 	rm -Rf "$TERMUX_TOPDIR/$TERMUX_PKG_NAME/subpackages"
-	for subpackage in $TERMUX_PKG_BUILDER_DIR/*.subpackage.sh $TERMUX_PKG_TMPDIR/*subpackage.sh; do
-		[[ -f "$subpackage" ]] || continue
+	for subpackage in "${TERMUX_PKG_SUBPACKAGES_LIST[@]}"; do
+		if [ ! -f "$subpackage" ]; then
```

Comment:
Wouldn't this be a more concise version of that change?
```suggestion
		[[ -f "$subpackage" ]] || termux_error_exit "Failed to find subpackage build file \"$subpackage\" of package \"$TERMUX_PKG_NAME\""
```
I spent a lot of time and effort overhauling the subpackages scripts for readability.
- https://github.com/termux/termux-packages/pull/24146

Reply:
Check https://github.com/termux/termux-packages/blob/63cfe26da5f26b1e8c54a689d68bc9f7d470ad81/scripts/build/termux_validate_built_packages.sh

We need to know if deb file for package and all its subpackages actually exists in the output directory before deciding whether package should be rebuilt or not. If any deb is missing, package must be rebuilt. Currently, there are no such validations.

Reply:
Sorry, check https://github.com/termux/termux-packages/blob/63cfe26da5f26b1e8c54a689d68bc9f7d470ad81/scripts/build/termux_set_subpackages.sh

That part was copied to a separate function due to duplicated code in debian and pacman.